src/app/api/widget-execution/route.tsx

```diff
@@ -0,0 +1,133 @@
+ * Step 4 - Route execution
+ *
+ * Example:
+ * ```
+ * http://localhost:3000/api/widget-execution?fromToken=0x0000000000000000000000000000000000000000&fromChainId=137&toToken=0x0000000000000000000000000000000000000000&toChainId=42161&amount=10&&theme=light&isSwap=true
+ * ```
+ *
+ * @typedef {Object} SearchParams
+ * @property {string} fromToken - The token address to send from.
+ * @property {number} fromChainId - The chain ID to send from.
+ * @property {string} toToken - The token address to send to.
+ * @property {number} toChainId - The chain ID to send to.
+ * @property {number} amount - The amount of tokens.
+ * @property {number} [amountUSD] - The USD equivalent amount (optional).
+ * @property {boolean} [isSwap] - True if transaction is a swap, default and false if transaction is a bridge (optional).
+ * @property {'light'|'dark'} [theme] - The theme for the widget (optional).
+ * @property {'from'|'to'|'amount'} [highlighted] - The highlighted element (optional).
+ *
+ */
+
+import type { ChainId } from '@lifi/sdk';
+import { ChainType, getChains, getToken } from '@lifi/sdk';
+import { ImageResponse } from 'next/og';
+import type { HighlightedAreas } from 'src/components/ImageGeneration/ImageGeneration.types';
+import { imageResponseOptions } from 'src/components/ImageGeneration/imageResponseOptions';
+import WidgetExecutionSSR from 'src/components/ImageGeneration/WidgetExecutionSSR';
+
+const WIDGET_IMAGE_WIDTH = 416;
+const WIDGET_IMAGE_HEIGHT = 432;
+const WIDGET_IMAGE_SCALING_FACTOR = 2;
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const amount = searchParams.get('amount');
+  const fromToken = searchParams.get('fromToken');
+  const fromChainId = searchParams.get('fromChainId');
+  const toToken = searchParams.get('toToken');
+  const toChainId = searchParams.get('toChainId');
+  const highlighted = searchParams.get('highlighted');
+  const theme = searchParams.get('theme');
+  const isSwap = searchParams.get('isSwap');
+
+  // Fetch chain data asynchronously before rendering
+  const getChainData = async (chainId: ChainId) => {
+    const chainsData = await getChains({
+      chainTypes: [ChainType.EVM, ChainType.SVM],
+    });
+    return chainsData.find((chainEl) => chainEl.id === chainId);
+  };
+
+  // Fetch from and to chain details (await this before rendering)
+  const fromChain = fromChainId
+    ? await getChainData(parseInt(fromChainId) as ChainId)
+    : null;
+  const toChain = toChainId
+    ? await getChainData(parseInt(toChainId) as ChainId)
+    : null;
+
+  // Fetch token asynchronously
+  const fetchToken = async (chainId: ChainId | null, token: string | null) => {
+    if (!chainId || !token) {
+      return null;
+    }
+    try {
+      const fetchedToken = await getToken(chainId, token);
+      return fetchedToken;
+    } catch (error) {
+      console.error('Error fetching token:', error);
+      return null;
+    }
+  };
```

Comment:
Mainly the same across endpoints, can be refactored

Reply:
Great idea!